pandas/io/json/json.py

```diff
@@ -600,15 +609,19 @@ class Parser(object):
         'us': long(31536000000000),
         'ns': long(31536000000000000)}
 
-    def __init__(self, json, orient, dtype=True, convert_axes=True,
+    def __init__(self, json, orient, dtype=None, convert_axes=True,
                  convert_dates=True, keep_default_dates=False, numpy=False,
                  precise_float=False, date_unit=None):
         self.json = json
 
         if orient is None:
             orient = self._default_orient
-
         self.orient = orient
+
```

Comment:
why are you doing this here?

Reply:
same comment as above

Reply:
write this like
``self.dtype = orient != 'table' if dtype is None else dtype``

Reply:
though still not sure why you cannot actually just set dtype when this is called with orient = table (from above)

Reply:
@jreback because that is precisely the aim of orient=table: the JSON object specifies itself its dtypes by means of the 'schema'.

Reply:
BTW, @jreback where could I have a more detailed discussion about the current implementation of pandas JSON serialization? I have found some issues related to JSON spec compliance.

Reply:
If your list is very exhaustive I'd suggest sending to the mailing list for back and forth. If it's something that can be summarized in a few checkpoints then an issue would be best

Reply:
@albertvillanova right, but my point is that we are now checking for the dtype above in read_json, then essentially checking again here. Why cannot we just pass it thru (after its checked once)?

Reply:
if you have other things you think should be fixed  pls open a new issue

Reply:
@jreback All default parameter settings (other than those defined in the signature) and inter-dependent constraints are defined at the instantiation of Parser: the default `orient` depending on the self type; `numpy=False` if `orient` is 'split'. That is the reason I put also there the default `dtype` depending on the `orient`.

Therefore, for consistency I would suggest either keeping all default settings as they are here in the Parser, or moving all of them to `read_json`.

Reply:
@jreback I have finally moved the setting of `dtype` to `read_json`, as you suggested.